DHFPROD-7681: Component Replacement: DatePicker

mrodrigu-marklogic opened the pull request:
### Description

#### Checklist: 
```diff
- Note: do not change the below
```

-  ##### Owner:

- [x] JIRA_ID included in all the commit messages
- [x] PR title is in the format JIRA_ID:Title
- [x] Rebase the branch with upstream
- [x] Squashed all commits into a single commit
- [x] Code passes ESLint tests
- [x] Added Tests
- [x] Run UI tests
  

- ##### Reviewer:

- [x] Reviewed Tests

federicoRufrancosMarklogic reviewed (dismissed):
It's approve, just one thing I notice, could font-family be inherited from the theme? Instead of using a value, just inherit

wooldridge commented on marklogic-data-hub-central/ui/src/components/common/hc-datetime-picker/hc-datetime-picker.module.scss:
```diff
@@ -0,0 +1,37 @@
+    right: 8px;
+    color: $light;
+    background-color: white;
+    font-size: 12px;
```
Can we please add the following to .calendarIcon since it is clickable:

cursor: pointer;

wooldridge reviewed (changes requested):
I cannot select a date range for a faceted date property.

1. In Modeling, make the NamespacedCustomer dateOfBirth property faceted.
2. In Explore, try to set a date range for NamespacedCustomer dateOfBirth property. It doesn't work.

Screencast: https://watch.screencastify.com/v/fVbqlLKOwzSWFkopjCrV


The error handling when applying the DatePicker Custom tag is a problem.

1. In Explore All Entities, click Custom from the createdOn menu. 
2. Don't select a date.
3. Apply the Custom tag to the search. I get a "date range is missing" error (which is OK).
4. Click OK. The error dialog keeps appearing. I expect the dialog to close.
5. Click Cancel. I get a 404 page error. I expect the dialog to close.

Screencast: https://watch.screencastify.com/v/ioIXCf9cZucGZFOyrLfD

ngodugu-marklogic commented:
@mrodrigu-marklogic Seems like the tests are failing because of 500 internal server error, where the tests are unable to interact with localhost:8080. I am not sure if this is an environment issue as I see the other PRs going fine. Should be specific to this PR. Tagging @SameeraPriyathamTadikonda to verify if this is an environment issue. And @mrodrigu-marklogic seems like there is a conflict. Please fix it.

SameeraPriyathamTadikonda commented:
I don't see any issues with the environment.

mrodrigu-marklogic commented:
@ngodugu-marklogic thank you for your heads up. Just rebased the branch and updated the PR to address some of the issues mentioned earlier in the QA review and will take a closer look on the test run results.

@wooldridge I found out that the issue of applying the empty Date facet has been happening on the Nightly build. Should we create a new ticket or does this to be addressed as a part of this Component Replacement PR? 

https://user-images.githubusercontent.com/88854466/135122220-d5f2c8ae-0faa-413b-9ea7-8435287ab1b7.mp4

Thanks in advance!

wooldridge commented:
> @ngodugu-marklogic thank you for your heads up. Just rebased the branch and updated the PR to address some of the issues mentioned earlier in the QA review and will take a closer look on the test run results.
> 
> @wooldridge I found out that the issue of applying the empty Date facet has been happening on the Nightly build. Should we create a new ticket or does this need to be addressed as a part of this Component Replacement PR?
> 
>  errorNightlyDate.mp4 
> Thanks in advance!

@mrodrigu-marklogic thanks for checking nightly. I will file a separate bug for that problem then. Don't worry about fixing the empty Data facet issue as part of this.
